torchgeo/datasets/geo.py

```diff
@@ -578,6 +579,93 @@ def __str__(self) -> str:
     size: {len(self)}"""
 
 
+class VisionClassificationDataset(VisionDataset, ImageFolder):  # type: ignore[misc]
+    """Abstract base class for classification datasets lacking geospatial information.
+
+    This base class is designed for datasets with pre-defined image chips which
+    are separated into separate folders per class.
+    """
+
+    def __init__(
+        self,
+        root: str,
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        loader: Optional[Callable[[str], Any]] = None,
+    ) -> None:
+        """Initialize a new VisionClassificationDataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            loader: a callable function which takes as input a path to an image and
+                returns a PIL Image or numpy array
+        """
+        # When transform & target_transform are None, ImageFolder.__getitem__[index]
+        # returns a PIL.Image and int for image and label, respectively
+        super().__init__(
+            root=root,
+            transform=None,
+            target_transform=None,
+            loader=default_loader if loader is None else loader,
+        )
+
+        # Must be set after calling super().__init__()
+        self.transforms = transforms
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+        Returns:
+            data and label at that index
+        """
+        image, label = self._load_image(index)
+        sample = {"image": image, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.imgs)
+
+    def __str__(self) -> str:
```

Comment:
We don't override `__str__` in `RasterDataset`, so I don't think we need to override it here either. Mostly users won't care if their dataset belongs to `VisionDataset` or `VisionClassificationDataset`, they're used the same way.